### PR TITLE
update lr scheduler doc for doing per step or epoch update

### PR DESCRIPTION
**Pull request by cli99**
DeepSpeed automatically updates the learning rate at every training step. This PR highlights this and describe what to do in different cases.

### CODE REVIEW

File: docs/_tutorials/getting-started.md [hunk 1]
```diff
@@ -26,10 +28,10 @@ model_engine, optimizer, _, _ = deepspeed.initialize(args=cmd_args,
 
 `deepspeed.initialize` ensures that all of the necessary setup required for
 distributed data parallel or mixed precision training are done
-appropriately under the hood.  In addition to wrapping the model, DeepSpeed can
+appropriately under the hood. In addition to wrapping the model, DeepSpeed can
 construct and manage the training optimizer, data loader, and the learning rate
 scheduler based on the parameters passed to `deepspeed.initialize` and the
-DeepSpeed [configuration file](#deepspeed-configuration).
+DeepSpeed [configuration file](#deepspeed-configuration). Note that DeepSpeed automatically updates the learning rate at every training step.
```
Comment (tjruwase) [inline docs/_tutorials/getting-started.md:34]: It is more precise that DeepSpeed executes the learning rate scheduler at every training step. The distinction is that a given lr_scheduler.step() may or may not change  the learning rate depending on the scheduler algorithm.

File: docs/_tutorials/getting-started.md [hunk 2]
```diff
@@ -66,32 +67,31 @@ for step, batch in enumerate(data_loader):
...
   gradients.
 
-* **Learning Rate Schedule**: if using DeepSpeed's learning rate
-  schedule, then DeepSpeed automatically handles any updates to the learning
-  rate when `step` is executed.
-
-
+- **Learning Rate Scheduler**: when using a DeepSpeed's learning rate scheduler (specified in the `ds_config.json` file), DeepSpeed automatically updates the learning rate at every training step (when `model_engine.step()` is executed). When not using a DeepSpeed's learning rate scheduler:
+  - if the scheduler is supposed to be updated every training step, then the user can pass the scheduler to `deepspeed.initialize` when initializing the DeepSpeed engine and let DeepSpeed manage it for update or save/restore.
+  - if the scheduler is supposed to be updated every training epoch, then the user should NOT pass the scheduler to DeepSpeed during initialization and must manage it explicitly.
```
Comment (tjruwase) [inline docs/_tutorials/getting-started.md:84]: This case should cover not just update at training epoch intervals, but any arbitrary intervals different from training step. So can we rewrite like, "if the schedule is supposed to be updated at any other interval (e.g., training epochs), ..."?

File: docs/code-docs/source/schedulers.rst [hunk 3]
```diff
@@ -1,8 +1,9 @@
 Learning Rate Schedulers
 ===================
 
-DeepSpeed offers implementations of ``LRRangeTest``, ``OneCycle``, ``WarmupLR``, ``WarmupDecayLR`` learning rate schedulers.
-
+DeepSpeed offers implementations of ``LRRangeTest``, ``OneCycle``, ``WarmupLR``, ``WarmupDecayLR`` learning rate schedulers.When using a DeepSpeed's learning rate scheduler (specified in the `ds_config.json` file), DeepSpeed automatically updates the learning rate at every training step (when `model_engine.step()` is executed). When not using a DeepSpeed's learning rate scheduler:
+  * if the scheduler is supposed to be updated every training step, then the user can pass the scheduler to `deepspeed.initialize` when initializing the DeepSpeed engine and let DeepSpeed manage it for update or save/restore.
```
Comment (tjruwase) [inline docs/code-docs/source/schedulers.rst:5]: Similar to above, DeepSpeed only calls the step() method of the scheduler, whether or not learning is actually updated depends on the specific scheduler algorithm.

File: docs/code-docs/source/schedulers.rst [hunk 4]
```diff
@@ -1,8 +1,9 @@
 Learning Rate Schedulers
 ===================
 
-DeepSpeed offers implementations of ``LRRangeTest``, ``OneCycle``, ``WarmupLR``, ``WarmupDecayLR`` learning rate schedulers.
-
+DeepSpeed offers implementations of ``LRRangeTest``, ``OneCycle``, ``WarmupLR``, ``WarmupDecayLR`` learning rate schedulers.When using a DeepSpeed's learning rate scheduler (specified in the `ds_config.json` file), DeepSpeed automatically updates the learning rate at every training step (when `model_engine.step()` is executed). When not using a DeepSpeed's learning rate scheduler:
+  * if the scheduler is supposed to be updated every training step, then the user can pass the scheduler to `deepspeed.initialize` when initializing the DeepSpeed engine and let DeepSpeed manage it for update or save/restore.
+  * if the scheduler is supposed to be updated every training epoch, then the user should NOT pass the scheduler to DeepSpeed during initialization and must manage it explicitly.
```
Comment (tjruwase) [inline docs/code-docs/source/schedulers.rst:6]: `if the scheduler is supposed to execute at any other interval besides training step (e.g., training epoch), then  ...`

**Review by tjruwase — APPROVED** [review]

File: docs/_tutorials/getting-started.md (hunk 1 above)
Comment (cli99) [reply docs/_tutorials/getting-started.md]: That's right.  Made the change. Thanks.

File: docs/_tutorials/getting-started.md (hunk 2 above)
Comment (cli99) [reply docs/_tutorials/getting-started.md]: Addressed.

File: docs/code-docs/source/schedulers.rst (hunk 3 above)
Comment (cli99) [reply docs/code-docs/source/schedulers.rst]: Addressed.

File: docs/code-docs/source/schedulers.rst (hunk 4 above)
Comment (cli99) [reply docs/code-docs/source/schedulers.rst]: Addressed.

**Comment by tjruwase** [comment]
@cli99 Thanks for this. Please merge at your convenience.
